game-core/src/main/java/games/strategy/triplea/formatter/MyFormatter.java

```diff
@@ -407,4 +408,38 @@ public static String integerUnitMapToString(
     }
     return buf.toString().replaceFirst(separator, "");
   }
+
+  /**
+   * Adds HTML line breaks and indentation to a string so it wraps for things like long tooltips.
+   *
+   * <pre>
+   * string part 1
+   *           string part 2
+   *           ...
+   *           string part X
+   * </pre>
+   */
+  public static String addHtmlBreaksAndIndents(
+      final String target, final int firstLineMaxLength, final int maxLength) {
+    final StringBuilder sb = new StringBuilder();
+    final BreakIterator breakIterator = BreakIterator.getLineInstance();
+    breakIterator.setText(target);
+    int start = breakIterator.first();
+    int end = breakIterator.next();
+    int lineLength = 0;
+    int currentMaxLength = firstLineMaxLength;
+    while (end != BreakIterator.DONE) {
+      final String word = target.substring(start, end);
+      lineLength = lineLength + word.length();
+      if (lineLength >= currentMaxLength) {
+        sb.append("<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;");
+        lineLength = word.length() + 5; // Add 5 for the indent
+        currentMaxLength = maxLength;
+      }
+      sb.append(word);
+      start = end;
+      end = breakIterator.next();
+    }
+    return sb.toString();
+  }
```

Comment:
I really have no clue what exact rules the logic here follows.
The Javadoc helps a bit, but the word wrapping makes stuff a little confusing.
Would you mind adding a few unit tests that test the behavior of this method an test edge cases with very long words?

Reply:
Yeah, I'll add a quick unit test though it follows the line breaks functionality provided by Java BreakIterator:
https://docs.oracle.com/javase/8/docs/api/java/text/BreakIterator.html#getLineInstance--
https://docs.oracle.com/javase/tutorial/i18n/text/line.html

Reply:
Added.